File: evidence/generated-systems/angular/governance-on-angular/src/app/models/Obligation.js

// Define collection and schema for Obligation
export interface Obligation {
    referenceNumber:
	type : string
    descriptionText:
	type : string
    Regulation:
	type : Schema.Types.ObjectId
    Controls:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Control' }]
    Policies:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Policy' }]
    Contracts:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Contract' }]
    ObligationType:
 	type : String
    ReviewFrequency:
 	type : String
#
    collection: 'obligations'
}

File: evidence/generated-systems/django/governance-on-django/governanceOnDjango/models/Obligation.py
from django.db import models
from governanceOnDjango.models.ObligationType import ObligationType
from governanceOnDjango.models.ControlFrequency import ControlFrequency

#======================================================================
# 
# Encapsulates data for model Obligation
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class Obligation Declaration
#======================================================================
class Obligation (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	referenceNumber = models.CharField(max_length=200, null=True)
	descriptionText = models.CharField(max_length=200, null=True)
	regulation = models.ForeignKey('Regulation', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	controls = models.ManyToManyField('Control',  blank=True, related_name='+')
	policies = models.ManyToManyField('Policy',  blank=True, related_name='+')
	contracts = models.ManyToManyField('Contract',  blank=True, related_name='+')
	obligationType = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in ObligationType])
	reviewFrequency = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in ControlFrequency])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.referenceNumber
		str = str + self.descriptionText
		str = str + self.obligationType
		str = str + self.reviewFrequency
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "Obligation";
    
	def objectType(self):
		return "Obligation";
